### PR TITLE
GLTF loading dirty mechanism perf

### PR DESCRIPTION
**Pull request by sebavan**
Lets try to not dirty the entire scene if possible when only adding new nodes/materials without changing the current state of the scene.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14487/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14487/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14487/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14487/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14487/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14487/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14487/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14487/merge#BCU1XR#0

File: packages/dev/core/src/Materials/Textures/baseTexture.ts [hunk 1]
```diff
@@ -319,9 +321,17 @@ export class BaseTexture extends ThinTexture implements IAnimatable {
...
     public set isRGBD(value: boolean) {
+        if (value === this.isRGBD) {
+            return;
+        }
+
         if (this._texture) {
             this._texture._isRGBD = value;
         }
+
+        this.getScene()?.markAllMaterialsAsDirty(Constants.MATERIAL_TextureDirtyFlag, (mat) => {
```
Comment (bghgary) [inline packages/dev/core/src/Materials/Textures/baseTexture.ts:332]: Maybe this should be inside the `if` above?

**Review by bghgary — APPROVED** [review]

File: packages/dev/core/src/Materials/Textures/baseTexture.ts (hunk 1 above)
Comment (sebavan) [reply packages/dev/core/src/Materials/Textures/baseTexture.ts:333]: It is like this for consistency with other places doing this

File: packages/dev/core/src/Materials/Textures/renderTargetTexture.ts [hunk 2]
```diff
@@ -498,6 +504,7 @@ export class RenderTargetTexture extends Texture implements IRenderTargetTexture
             noColorAttachment = !!options.noColorAttachment;
             useSRGBBuffer = !!options.useSRGBBuffer;
             colorAttachment = options.colorAttachment;
+            gammaSpace = !!options.gammaSpace;
```
Comment (kircher1) [inline packages/dev/core/src/Materials/Textures/renderTargetTexture.ts:507]: default gammaSpace to true?

File: packages/dev/core/src/Materials/Textures/texture.ts [hunk 3]
```diff
@@ -420,10 +425,12 @@ export class Texture extends BaseTexture {
             creationFlags = noMipmapOrOptions.creationFlags;
             useSRGBBuffer = noMipmapOrOptions.useSRGBBuffer ?? false;
             internalTexture = noMipmapOrOptions.internalTexture ?? null;
+            gammaSpace = noMipmapOrOptions.gammaSpace ?? false;
```
Comment (kircher1) [inline packages/dev/core/src/Materials/Textures/texture.ts:428]: default true I think

File: packages/dev/core/src/Materials/Textures/texture.ts [hunk 4]
```diff
@@ -420,10 +425,12 @@ export class Texture extends BaseTexture {
             creationFlags = noMipmapOrOptions.creationFlags;
             useSRGBBuffer = noMipmapOrOptions.useSRGBBuffer ?? false;
             internalTexture = noMipmapOrOptions.internalTexture ?? null;
+            gammaSpace = noMipmapOrOptions.gammaSpace ?? false;
         } else {
             noMipmap = !!noMipmapOrOptions;
         }
 
+        this._gammaSpace = !!gammaSpace;
```
Comment (kircher1) [inline packages/dev/core/src/Materials/Textures/texture.ts:433]: default true

File: packages/dev/core/src/Materials/Textures/renderTargetTexture.ts (hunk 2 above)
Comment (sebavan) [reply packages/dev/core/src/Materials/Textures/renderTargetTexture.ts:507]: ohhhh yes thanks a lot.....

File: packages/dev/core/src/Materials/Textures/renderTargetTexture.ts [hunk 5]
```diff
@@ -457,7 +461,8 @@ export class RenderTargetTexture extends Texture implements IRenderTargetTexture
         samples?: number,
         creationFlags?: number,
         noColorAttachment?: boolean,
-        useSRGBBuffer?: boolean
+        useSRGBBuffer?: boolean,
+        gammaSpace?: boolean
```
Comment (Popov72) [inline packages/dev/core/src/Materials/Textures/renderTargetTexture.ts:465]: I think that now that we have a `RenderTargetTextureOptions` object parameter, we shouldn't add new parameters in this way (that's what we did for the last option - `colorAttachment` - that we added)?

File: packages/dev/core/src/Materials/Textures/renderTargetTexture.ts [hunk 6]
```diff
@@ -478,7 +483,8 @@ export class RenderTargetTexture extends Texture implements IRenderTargetTexture
         samples?: number,
         creationFlags?: number,
         noColorAttachment = false,
-        useSRGBBuffer = false
+        useSRGBBuffer = false,
+        gammaSpace = true
```
Comment (Popov72) [inline packages/dev/core/src/Materials/Textures/renderTargetTexture.ts:487]: Same.

File: packages/dev/core/src/Materials/Textures/texture.ts [hunk 7]
```diff
@@ -395,7 +399,8 @@ export class Texture extends BaseTexture {
         mimeType?: string,
         loaderOptions?: any,
         creationFlags?: number,
-        forcedExtension?: string
+        forcedExtension?: string,
+        gammaSpace = true
```
Comment (Popov72) [inline packages/dev/core/src/Materials/Textures/texture.ts:403]: Same.

File: packages/dev/core/src/Materials/Textures/renderTargetTexture.ts (hunk 5 above)
Comment (sebavan) [reply packages/dev/core/src/Materials/Textures/renderTargetTexture.ts]: yup totally agree I ll keep only the option ones.

**Review by Popov72 — APPROVED** [review]

**Pull request merged**